diff --git a/core/src/impl/Kokkos_SharedAlloc.hpp b/core/src/impl/Kokkos_SharedAlloc.hpp
index 480ad1691..ccd92f811 100644
--- a/core/src/impl/Kokkos_SharedAlloc.hpp
+++ b/core/src/impl/Kokkos_SharedAlloc.hpp
@@ -614,12 +614,12 @@ union SharedAllocationTracker {
 
   KOKKOS_FORCEINLINE_FUNCTION
 #if defined(KOKKOS_COMPILER_NVCC) || !defined(KOKKOS_COMPILER_GNU) || \
-    (KOKKOS_COMPILER_GNU < 1220) || (KOKKOS_COMPILER_GNU > 1240)
-      // FIXME_GCC: The ViewSupport test fails with gcc 12.2, 12.3 and 12.4
+    (KOKKOS_COMPILER_GNU < 1220) || (KOKKOS_COMPILER_GNU > 1250)
+      // FIXME_GCC: The ViewSupport test fails with gcc 12.2-12.5
       // because this constructor is optimized out, which leads to a nullptr
       // dereference. Removing the constexpr fixes the issue but nvcc complains,
       // so we keep the constexpr but only when using anything other than those
-      // three faulty gcc versions.
+      // four faulty gcc versions.
       constexpr
 #endif
       SharedAllocationTracker()
